test/jdk/java/awt/Robot/ScreenCaptureRobotTest.java

```diff
@@ -0,0 +1,147 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Canvas;
+import java.awt.Color;
+import java.awt.EventQueue;
+import java.awt.Font;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.GraphicsEnvironment;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.image.BufferedImage;
+
+/*
+ * @test
+ * @key headful
+ * @bug 8342098
+ * @summary Verify that the image captured from the screen using a Robot
+ * and the source image are same.
+ * @run main ScreenCaptureRobotTest
+ */
+public class ScreenCaptureRobotTest {
+
+    private static int delay = 500;
+    private static Frame frame;
+    private static volatile Canvas canvas;
+    private static BufferedImage realImage;
+    private static BufferedImage displayImage;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            EventQueue.invokeAndWait(ScreenCaptureRobotTest::initializeGUI);
+            doTest();
+        } finally {
+            EventQueue.invokeAndWait(ScreenCaptureRobotTest::disposeFrame);
+        }
+    }
+
+    private static void initializeGUI() {
+        frame = new Frame("ScreenCaptureRobotTest Frame");
+        realImage = GraphicsEnvironment.getLocalGraphicsEnvironment()
+                .getDefaultScreenDevice().getDefaultConfiguration()
+                .createCompatibleImage(200, 100);
+
+        Graphics g = realImage.createGraphics();
+        g.setColor(Color.yellow);
+        g.fillRect(0, 0, 200, 100);
+        g.setColor(Color.red);
+        g.setFont(new Font("SansSerif", Font.BOLD, 20));
+        g.drawString("Capture This", 10, 40);
+        g.dispose();
+        displayImage = realImage;
+
+        canvas = new ImageCanvas();
+        canvas.setBackground(Color.YELLOW);
+        frame.setLayout(new BorderLayout());
+        frame.add(canvas);
+        frame.setSize(300, 200);
+        frame.setLocation(100, 100);
+        frame.setVisible(true);
+        canvas.requestFocus();
+    }
+
+    private static void doTest() throws Exception {
+        Robot robot;
+        robot = new Robot();
+        robot.delay(delay);
+        robot.waitForIdle();
+
+        Point pnt = canvas.getLocationOnScreen();
+        Rectangle rect = new Rectangle(pnt.x + 10, pnt.y + 10, 200, 100);
```

Comment:
Why are coordinates of the start of the canvas are offset by 10 pixels but the size isn't. The image is 200×100, then you capture 10 pixels on the right and bottom which belong to whatever is on the screen in this area.

Reply:
@aivanov-jdk 
While creating the Canvas 
drawImage(realImage, 10, 10, this)
10 offset is used.

Hence the start point coordinates also moved by 10 pixels
for capturing.

Reply:
Okay, I see that you offset the image on the canvas too.

If you insist on using an offset, please declare a constant `OFFSET = 10` and use it everywhere instead of the ‘magic’ numbers: when painting to canvas, and when capturing the image.

You used a constant for a delay that was used once but you don't create a constant to offset the image off edges that you use in two different places. If one place is updated, the other has to be updated as well. Using a constant does it all automatically.

Reply:
> @aivanov-jdk While creating the Canvas drawImage(realImage, 10, 10, this) 10 offset is used.
> 
> Hence the start point coordinates also moved by 10 pixels for capturing.

This is why _you should create a constant_ for this offset and use when drawing the image on the canvas and when capturing the screen.